lib/polygon-feature.js

```diff
@@ -0,0 +1,52 @@
+var forEach = require('lodash/forEach')
+
+module.exports = polygonFeature
+
+function polygonFeature (presets) {
+  var areaKeys = {}
+  var ignore = ['barrier', 'highway', 'footway', 'railway', 'type']
+
+  // whitelist
+  forEach(presets, function (d) {
+    for (var key in d.tags) break
+    if (!key) return
+    if (ignore.indexOf(key) !== -1) return
+
+    if (d.geometry.indexOf('area') !== -1) {
+      areaKeys[key] = areaKeys[key] || {}
+    }
+  })
+
+  // blacklist
+  forEach(presets, function (d) {
+    for (var key in d.tags) break
+    if (!key) return
+    if (ignore.indexOf(key) !== -1) return
+
+    var value = d.tags[key]
+    if (d.geometry.indexOf('area') === -1 &&
+                d.geometry.indexOf('line') !== -1 &&
+                key in areaKeys && value !== '*') {
+      areaKeys[key][value] = true
+    }
+  })
+
+  return function isPolygon (coords, tags) {
```

Comment:
This is cut & pasted from https://github.com/openstreetmap/iD/blob/9b7d79a3fec6f0eb5e292c3b89a2912fd8751197/modules/presets/index.js#L62-L104